api/schema/k8s_cluster.go

```diff
@@ -0,0 +1,71 @@
+
+package schema
+
+import (
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+)
+
+var (
+	k8sClusterResource = Resource{
+		TypeName: "k8s_cluster",
+		Schema: resource_schema.Schema{
+			Version:             1,
+			MarkdownDescription: "Manages a Kubernetes cluster that can onboarded/offboarded to CloudSecure within a specific Illumio Region.",
+			Attributes: map[string]resource_schema.Attribute{
+				IDFieldName: idAttribute,
+				"client_id": StringResourceAttributeWithMode{
+					StringAttribute: resource_schema.StringAttribute{
+						MarkdownDescription: "Client identifier used by CloudSecure's k8s operator to authenticate to CloudSecure for onboarding/offboarding, in combination with `client_secret`. Identical to `id`.",
+						Computed:            true,
+					},
+					attributeWithMode: attributeWithMode{
+						Mode: ReadOnlyAttributeMode,
+					},
+				},
+				"client_secret": StringResourceAttributeWithMode{
+					StringAttribute: resource_schema.StringAttribute{
+						MarkdownDescription: "Client secret used by CloudSecure's k8s operator to authenticate to CloudSecure for onboarding/offboarding, in combination with `client_id`.",
+						Computed:            true,
+						Sensitive:           true,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
+					},
+					attributeWithMode: attributeWithMode{
+						Mode: ReadOnlyOnceAttributeMode,
+					},
+				},
+				"illumio_region": StringResourceAttributeWithMode{
+					StringAttribute: resource_schema.StringAttribute{
+						MarkdownDescription: "Illumio Region where the k8s cluster will be onboarded or offboarded from" +
+							"An Illumio Region is a designated cloud region where the CloudSecure k8s operators in onboarded k8s clusters connect after onboarding. " +
+							"Choose the Illumio Region nearest to each cluster to maximize performance and security. " +
+							"Must be one of: `aws-ap-southeast-2`, `aws-eu-west-2`, `aws-us-west-2`.",
+						Required: true,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.RequiresReplace(),
+						},
+					},
+					attributeWithMode: attributeWithMode{
+						Mode: ImmutableAttributeMode,
+					},
+				},
+				"log_level": StringResourceAttributeWithMode{
+					StringAttribute: resource_schema.StringAttribute{
+						Optional: true,
+						MarkdownDescription: "Log level indicates what verbosity of logs the cloud operator will print out. " +
+							"This configuration option may become of use if CloudSecure support needs to observe and debug issues on a given cluster. " +
+							"Must be one of: `debug`, `info`, `warn`, or `error`.",
+					},
+					attributeWithMode: attributeWithMode{
+						Mode: ReadWriteAttributeMode,
+					},
+				},
```

Comment:
1. Capitalize the enum values, to be consistent with the other resources ("Allow", "Deny") etc.
1. Add a default.
1. Add a validator.
1. `ReadWriteAttributeMode` is the default, so no need to specify it.

See for example the `mode` attribute for `aws_account`: https://github.com/illumio/terraform-provider-illumio-cloudsecure/blob/3294e2240c4b22f05b3151d1312c13cd1cc9f0a7/api/schema/azure_subscription_resource.go#L47-L63
```suggestion
				"log_level": StringAttribute: resource_schema.StringAttribute{
					MarkdownDescription: "Verbosity of the logs produced by the CloudSecure k8s operator. " +
						"Must be one of: `Debug`, `Info`, `Warn`, or `Error`.",
					Optional:    true,
					Default:     stringdefault.StaticString("Info"),
					Validators: []validator.String{
						stringvalidator.OneOf("Debug", "Info", "Warn", "Error"),
					},
				},
```

Reply:
Gotcha okay the validator completely slipped my mind. Thanks for laying this out